File: app/src/main/java/com/shop440/Adapters/ProductAdapter.java
package com.shop440.Adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.shop440.Models.Datum;

import java.util.List;

/**
 * Created by SMILECS on 1/24/17.
 */

public class ProductAdapter extends RecyclerView.Adapter<ProductAdapter.ViewHolder>{
    List<Datum> model;
    Context c;
    public ProductAdapter(Context c, List<Datum> model){
        this.model = model;
        this.c = c;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{
        TextView StoreName, price, product, category, location;
        ImageView logo;
        public ViewHolder(final View itemView) {
            super(itemView);

        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {


    }

    @Override
    public int getItemCount() {
        return model.size();
    }
}

File: app/src/main/java/com/shop440/productview/adapter/ProductAdapter.kt
package com.shop440.productview.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.shop440.models.Datum

/**
 * Created by SMILECS on 1/24/17.
 */

class ProductAdapter(internal var c: Context, internal var model: List<Datum>) : RecyclerView.Adapter<ProductAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var StoreName: TextView? = null
        internal var price: TextView? = null
        internal var product: TextView? = null
        internal var category: TextView? = null
        internal var location: TextView? = null
        internal var logo: ImageView? = null
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder? {
        return null
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {


    }

    override fun getItemCount(): Int {
        return model.size
    }
}
